x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/MountSearchableSnapshotAction.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.searchablesnapshots.action;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.admin.cluster.snapshots.restore.RestoreSnapshotResponse;
+
+public class MountSearchableSnapshotAction extends ActionType<RestoreSnapshotResponse> {
+
+    public static final MountSearchableSnapshotAction INSTANCE = new MountSearchableSnapshotAction();
+    public static final String NAME = "cluster:admin/snapshot/mount";
```

Comment:
Should we make this uniform we other searchable snapshots related actions?

Reply:
I think it's either conform to the searchable snapshot actions, or treat it more like the snapshot action. Here I decided to treat it like a "snapshot" action (a "mount" action) so people could grant `cluster:admin/snapshot/*` and be able to mount searchable snapshots (since it's essentially a specialized restore)